R/Shiny.R

```diff
@@ -110,128 +110,59 @@ launchDiagnosticsExplorer <- function(dataFolder = "data",
+#' This function combines diagnostics results from one or more databases into a single file. The result is an
+#' sqlite database that can be used as input for the Diagnostics Explorer Shiny app.
 #'
 #' It also checks whether the results conform to the results data model specifications.
 #'
 #' @param dataFolder       folder where the exported zip files for the diagnostics are stored. Use
 #'                         the \code{\link{runCohortDiagnostics}} function to generate these zip files.
 #'                         Zip files containing results from multiple databases may be placed in the same
 #'                         folder.
-#' @param tempFolder       A folder on the local file system where the zip files are extracted to. Will be cleaned
-#'                         up when the function is finished. Can be used to specify a temp folder on a drive that
-#'                         has sufficient space if the default system temp space is too limited.
-#'
+#' @param sqliteDbPath     Output path where sqlite database is placed
+#' @param overwrite        (Optional) overwrite existing sqlite lite db if it exists.
 #' @export
-preMergeDiagnosticsFiles <-
-  function(dataFolder, tempFolder = tempdir()) {
-    zipFiles <-
-      dplyr::tibble(
-        zipFile = list.files(
-          dataFolder,
-          pattern = ".zip",
-          full.names = TRUE,
-          recursive = TRUE
-        ),
-        unzipFolder = ""
-      )
-    ParallelLogger::logInfo("Merging ", nrow(zipFiles), " zip files.")
-    
-    unzipMainFolder <-
-      tempfile("unzipTempFolder", tmpdir = tempFolder)
-    dir.create(path = unzipMainFolder, recursive = TRUE)
-    on.exit(unlink(unzipMainFolder, recursive = TRUE))
-    
-    for (i in 1:nrow(zipFiles)) {
-      ParallelLogger::logInfo("- Unzipping ", basename(zipFiles$zipFile[i]))
-      unzipFolder <-
-        file.path(unzipMainFolder, sub(".zip", "", basename(zipFiles$zipFile[i])))
-      dir.create(unzipFolder)
-      zip::unzip(zipFiles$zipFile[i], exdir = unzipFolder)
-      zipFiles$unzipFolder[i] <- unzipFolder
+createMergedResultsFile <-
+  function(dataFolder,
+           sqliteDbPath = "MergedCohortDiagnosticsData.sqlite",
+           overwrite = FALSE) {
+
+    if (file.exists(sqliteDbPath) & !overwrite) {
+      stop("File ", sqliteDbPath, " already exists. Set overwrite = TRUE to replace")
+    } else if (file.exists(sqliteDbPath)){
+      unlink(sqliteDbPath)
     }
-    
-    specifications <- getResultsDataModelSpecifications()
-    
-    # Storing output in an environment for now. If things get too big, we may want to write
-    # directly to CSV files for insertion into database:
-    newEnvironment <- new.env()
-    
-    processTable <- function(tableName, env) {
-      ParallelLogger::logInfo("Processing table ", tableName)
-      csvFileName <- paste0(tableName, ".csv")
-      data <- dplyr::tibble()
-      for (i in 1:nrow(zipFiles)) {
```

Comment:
@azimov - wow, all these checks will be gone because sqlite will enforce column names, data types and duplicate rows!